sentry_sdk/attachments.py

```diff
@@ -0,0 +1,51 @@
+import os
+import mimetypes
+
+from sentry_sdk._types import MYPY
+from sentry_sdk.envelope import Item, PayloadRef
+
+if MYPY:
+    from typing import Optional
+
+
+class Attachment(object):
+    def __init__(
+        self,
+        bytes=None,  # type: Optional[bytes]
+        filename=None,  # type: Optional[str]
+        path=None,  # type: Optional[str]
+        content_type=None,  # type: Optional[str]
+        add_to_transactions=False,  # type: bool
+    ):
+        # type: (...) -> None
+        if bytes is None and path is None:
+            raise TypeError("path or raw bytes required for attachment")
+        if filename is None and path is not None:
+            filename = os.path.basename(path)
+        if filename is None:
+            raise TypeError("filename is required for attachment")
```

Comment:
The main argument I think against conflating `path` and `filename` is that if it becomes the same, it's a dual responsibility for different things. `path` is the absolute path in OS native file format (bytes or unicode) where the file is read from and `filename` is the relative filename what the file is reported at for the UI (unicode only).

So for instance a not uncommon situation is that you are trying to attach a logfile that has a filename such as /var/log/application-log-20201010.log` and you want to report it as `logfile.txt`.

>  And, out of curiosity, why does an envelope item has a slot for a filename instead of a full path?

Because that's what the file is reported at in the UI. We intentionally do not want to keep an entire file path there as it causes more complexities and does not solve a purpose for the user.

Reply:
Going to stay with the current API.